packages/messenger-widget/src/components/MessageInputBox/MessageInputBox.tsx

```diff
@@ -56,12 +59,27 @@ export function MessageInputBox() {
         }
     }, [messageView]);
 
+    /**
+     * Updates contact selected locally on change on selected contact.
+     * Since the updatedAt property changes on every message, due to which input field of message
+     * gets cleared which results in loss of message written in input field if new message is received
+     * at the same time. So to avoid that contactSelected is used to track message input field is
+     * cleared only if selectedContact is changed and not on updatedAt property of selectedContact
+     * while chatting
+     */
     useEffect(() => {
-        setMessageView({
-            actionType: MessageActionType.NONE,
-            messageData: undefined,
-        });
-        setMessage('');
+        if (
+            !contactSelected ||
+            contactSelected.contactDetails.account.ensName !==
```

Comment:
isn't that always false ?

Reply:
would it be cleaner to fix the selectedContact in useConversation instead of having it redundant here ?

Reply:
But there also I need to have extra state to know whether contact selected is changed or just updatedAt property of contact is changed due to new message